packages/compiler/src/core/name-resolver.ts

```diff
@@ -0,0 +1,1297 @@
+  }
+
+  /**
+   * This method will take a member container and compute all the known member
+   * symbols. It will determine whether it has unknown members and set the
+   * symbol link value appropriately. This is used during resolution to know if
+   * member resolution should return `unknown` when a member isn't found.
+   */
+  function bindMemberContainer(node: Node) {
+    const sym = node.symbol!;
+    const symLinks = getSymbolLinks(sym);
+
+    if (symLinks.membersBound) {
+      return;
+    }
+
+    symLinks.membersBound = true;
+
+    switch (node.kind) {
+      case SyntaxKind.ModelStatement:
+      case SyntaxKind.ModelExpression:
+        bindModelMembers(node);
+        return;
+      case SyntaxKind.IntersectionExpression:
+        bindIntersectionMembers(node);
+        return;
+      case SyntaxKind.InterfaceStatement:
+        bindInterfaceMembers(node);
+        return;
+      case SyntaxKind.EnumStatement:
+        bindEnumMembers(node);
+        return;
+      case SyntaxKind.UnionStatement:
+        bindUnionMembers(node);
+        return;
+      case SyntaxKind.ScalarStatement:
+        bindScalarMembers(node);
+        return;
+    }
+  }
+
+  // TODO: had to keep the metaTypeMembers which this pr originally tried to get rid as we need for ops parameters to be cloned and have a new reference
+  function bindOperationStatementParameters(node: OperationStatementNode) {
+    const targetTable = getAugmentedSymbolTable(node.symbol!.metatypeMembers!);
+    if (node.signature.kind === SyntaxKind.OperationSignatureDeclaration) {
+      const { finalSymbol: sym } = resolveExpression(node.signature.parameters);
+      if (sym) {
+        targetTable.set("parameters", sym);
+      }
+    } else {
+      const { finalSymbol: sig } = resolveTypeReference(node.signature.baseOperation);
+      if (sig) {
+        const sigTable = getAugmentedSymbolTable(sig.metatypeMembers!);
+        const sigParameterSym = sigTable.get("parameters")!;
+        if (sigParameterSym !== undefined) {
+          const parametersSym = createSymbol(
+            sigParameterSym.node,
+            "parameters",
+            SymbolFlags.Model & SymbolFlags.MemberContainer,
+          );
+          getAugmentedSymbolTable(parametersSym.members!).include(
+            getAugmentedSymbolTable(sigParameterSym.members!),
+            parametersSym,
+          );
+          targetTable.set("parameters", parametersSym);
+          targetTable.set("returnType", sigTable.get("returnType")!);
+        }
+      }
+    }
+  }
+
+  function bindDeclarationIdentifier(node: Node & { id: IdentifierNode }) {
+    if (node.kind === SyntaxKind.TypeSpecScript || node.kind === SyntaxKind.JsSourceFile) return;
+    const links = getNodeLinks(node.id);
+    let sym;
+    if (node.symbol === undefined) {
+      return;
+    }
+    if (node.symbol.flags & SymbolFlags.Member) {
+      // TODO: we shouldn't probably have to get the parent, can we not do it when traversing down.
```

Comment:
This todo seems actionable, might just need to have a more specific bind declaration identifier